petastorm/spark/spark_dataset_converter.py

```diff
@@ -69,19 +70,32 @@ def _cache_df_or_retrieve_cache_path(df, cache_dir, row_group_size):
     """
     uuid_str = str(uuid.uuid4())
     save_to_dir = os.path.join(cache_dir, uuid_str)
-    df.write.mode("overwrite") \
+
+    old_compression = _spark_session.conf.get("spark.sql.parquet.compression.codec")
+
+    _spark_session.conf.set(
+        "spark.sql.parquet.compression.codec", compression)
+    df.write \
```

Comment:
Did you removed `df.write.mode("overwrite")` on purpose?

Reply:
We always create a new directory path (by naming uuid4), so no overwrite case.